platforms/core-runtime/client-services/src/main/java/org/gradle/launcher/daemon/toolchain/DaemonClientToolchainServices.java

```diff
@@ -115,15 +137,17 @@ protected Lazy<JavaToolchainQueryService> createJavaToolchainQueryService(
 
             CurrentBuildPlatform currentBuildPlatform = new CurrentBuildPlatform(systemInfo, os);
             DefaultFilePropertyFactory filePropertyFactory = new DefaultFilePropertyFactory(propertyHost, fileResolver, fileCollectionFactory);
-            JdkCacheDirectory jdkCacheDirectory = new DefaultJdkCacheDirectory(gradleUserHomeDirProvider, null, fileLockManager, new DefaultJvmMetadataDetector(execHandleFactory, gradleUserHomeTemporaryFileProvider), gradleUserHomeTemporaryFileProvider);
+            DecompressionCoordinator decompressionCoordinator = new DefaultDecompressionCoordinator(scopedCacheBuilderFactory);
```

Comment:
This service ends up being needed and we cannot use an implementation that does not handle conflicts. However I still need to confirm that used from the launcher, this would be safe to use as-is.

Reply:
If not wrong I suppose that the reason of not using `DefaultDecompressionCoordinator ` was that at the time I implemented probably `ScopedCacheBuilderFactory` wasn't accessible that's why I end up creating a basic `DecompressionCoordinator `.

Reply:
Confirmed that we can use this service as a coordinator